Tickets/dm 36146

pcortes80 opened the pull request:
Please check this PR.
Thanks.

pcortes80 committed:
Create code in switchRelays.vi and mainFPGA.vi.

pcortes80 committed:
Remove unnecesary directories.

pcortes80 committed:
Compile FPGA and generate FPGA C API Interface.

pcortes80 committed:
Update documents.

teweitsai commented on README.md:
```diff
@@ -5,6 +5,26 @@
+
+- Windows 10
+- LabVIEW 2018 SP1 32-bit
+- cRIO-9049 (M2 simulator: m2-crio-simulator.ls.lsst.org)
```
Please remove this IP address of `m2-crio-simulator.ls.lsst.org`.
We will remove this cRIO from base data center and put to the real M2 hardware soon.
The named IP address will be changed and can only be accessed by VPN at that time.

Thanks!

pcortes80 replied:
Done.

teweitsai commented on README.md:
```diff
@@ -5,6 +5,26 @@
+- LabVIEW NI Device Drivers (DVD-Jan-19-1)
+- LabVIEW 2018 FPGA Module
+- LabVIEW 2018 SP1 CompactRIO Module 18.5 (Jan, 2019)
+- LabVIEW 2018 FPGA Module Xilinx Compilation Tool for Vivado 2017.2
```
I think this is optional if you use the NI cloud to compile the FPGA. Isn't it?

pcortes80 replied:
You are right. Let me leave it with a comment (optional), because, if you don't want to use the NI cloud compile, it is good to know wich version to download.

teweitsai commented on README.md:
```diff
@@ -22,16 +42,26 @@ ts_m2fpga
-│   ├── NiFpga_mainFPGA.lvbitx
 │   ├── NiFpga.c
 │   ├── NiFpga.h
+│   ├── NiFpga_mainFPGA.lvbitx
+│   ├── bitfile
+│       ├── m2fpga_FPGATarget_mainFPGA_<random_characters>.lvbitx
```
Why do you have two `lvbitx` here?

pcortes80 replied:
NiFpga_mainFPGA.lvbitx is a version of the original bitfile generated after FPGA compilation. This copy will be added to version control. The fpgaInterface/bitfile directory will be added to the gitignore file.

pcortes80 replied:
**NiFpga_mainFPGA.lvbitx** is a version of the original bitfile created after FPGA compilation. So, the **fpgaInterface/bitfile** directory will be added to the **.gitignore** file because is not necessary to keep it in version control.

teweitsai commented on README.md:
```diff
@@ -22,16 +42,26 @@ ts_m2fpga
 - `src`: contains the LabVIEW FPGA project.
 - `doc`: documentation like `versionHistory.md`.
 - `fpgaInterface`: in here are all files generated by the FPGA Interface C API.
+- `bitfile`: contains the bitfile (.lvbitx).
```
`fpgaInterface/bitfile`: contains the bitfile (`.lvbitx`).

pcortes80 replied:
Done.

teweitsai commented on README.md:
```diff
@@ -22,16 +42,26 @@ ts_m2fpga
+When the FPGA application executes, the bitfile loads on the FPGA chip and reconfigures the gate array logic.
+Follow [this link](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000YHVTCA4&l=en-US) if you want to know more about the LabVIEW FPGA compilation process.
+
+Random characters in the `lvbitx` file name appear when the **mainFPGA.vi** is compiled.
```
I do not understand this sentence at all: "Random characters in the `lvbitx` file name".

pcortes80 replied:
I realized there is not necessary to explain these random characters, because if the developer compiles the FPGA again, the bitfile name will be the same.

teweitsai reviewed (approved):
Looks good. Please discuss first before any modification. Thanks!

[m2fpga.lvproj]

1. I would like to have the hierarchy of vis.

[mainFPGA.vi]

1. Should we put the Loop Timer here? And why it is 100 msec?
2. Do we need the error in and out?
3. For the ICON, do not use "all capitals".


[switchRelays.vi]

1. For the ICON, do not use "all capitals".
2. I do not understand the doc string of this vi.
3. For the controlRelays doc string, I think it is better not to identify the model name (NI-9485).
4. Do we need the error in and out?

pcortes80 commented:
[m2fpga.lvproj]

1. I would like to have the hierarchy of vis.
- this was done.

[mainFPGA.vi]

1. Should we put the Loop Timer here? And why it is 100 msec?
- I removed this Loop Timmer.

2. Do we need the error in and out?
- after discussion, we decided to add the error in and out.

3. For the ICON, do not use "all capitals".
- I just changed the ICON fonts to avoid using "all capitals".

[switchRelays.vi]

1. For the ICON, do not use "all capitals".
- I just changed the ICON fonts to avoid using "all capitals".

2. I do not understand the doc string of this vi.
- Improve this description.

3. For the controlRelays doc string, I think it is better not to identify the model name (NI-9485).
- got it, I improved this description

4. Do we need the error in and out?
- after discussion, we decided to add the error in and out.

pcortes80 committed:
Modify according to Te-Wei's comments.
